### ISSUE TITLE
Feature: Support regex in profile activation via kubeContext

### ISSUE THREAD

**Issue by dkirrane** [GoogleContainerTools/skaffold#1677]
It would be good to auto-activate a Skaffold profile based on a kubeContext regex.

e.g. My clusters (both gke & on-prem) have names like the following:

-  `gke_my-team_europe-west2-a_my-dev-cluster`
-  `gke_my-team_europe-west2-a_my-sanity-cluster`
- `kubernetes-admin@dev`
- `kubernetes-admin@sanity`

I want to run the gke profile below when my kube context matches a GKE cluster name (using regex `gke.*`)

```
apiVersion: skaffold/v1beta5
kind: Config
.
.
.
profiles:
  - name: gke
    build:
      artifacts:
        - image: eu.gcr.io/my-team/my-app
          jibMaven: {}
    activation:
      - kubeContext: "gke.*"
```

**Comment by charlyx** [comment]
`kubeContext` actually supports "equals" AND "not equals" by prefixing with `!`.
Shall we add a `kubeContextRegexp` so we don't mix things with current behaviour?

**Comment by dgageot** [comment]
Fixed with #2065
